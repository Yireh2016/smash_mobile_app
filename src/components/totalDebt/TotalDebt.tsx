import * as React from 'react';
import { View, StyleSheet } from 'react-native';

import { MainTitle, P2 } from '../../components/texts/Texts';

interface TotalDebtProps {
  totalDebt: string;
  title?: string;
}
const styles = StyleSheet.create({
  debtLayout: {
    marginTop: 34,
    alignItems: 'center',
  },
});
const TotalDebt: React.FunctionComponent<TotalDebtProps> = ({
  totalDebt,
  title = 'Total credit card debt',
}) => {
  return (
    <View style={styles.debtLayout}>
      <P2>{title}</P2>
      <MainTitle>{totalDebt}</MainTitle>
    </View>
  );
};

export default TotalDebt;
