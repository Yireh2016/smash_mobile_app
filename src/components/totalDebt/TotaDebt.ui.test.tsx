import * as React from 'react';
import renderer from 'react-test-renderer';
import TotalDebt from './TotalDebt';

it('TotalDebt renders correctly', () => {
  const Element = () => {
    return <TotalDebt totalDebt="$2,350.45" />;
  };

  const ElementRendered = renderer.create(<Element />).toJSON();
  expect(ElementRendered).toMatchSnapshot();
});
