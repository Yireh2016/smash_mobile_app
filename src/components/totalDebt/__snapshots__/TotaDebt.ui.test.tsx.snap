// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalDebt renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 34,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 14,
      }
    }
  >
    Total credit card debt
  </Text>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "OpenSans-Bold",
        "fontSize": 28,
      }
    }
  >
    $2,350.45
  </Text>
</View>
`;
